### PR TITLE
Lvnod 120 - Update profile page view

### PR DESCRIPTION
**Pull request by yuliannakorol**
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/browse/LVNOD-120)
* [JIRA subtask](https://ssu-jira.softserveinc.com/browse/LVNOD-121)

## Code reviewers

- [x] @yarvanya
- [ ] @yurii-kokhalevych
- [ ] @avoytovich

### Second Level Review

- [ ] @lhalam
- [ ] @4asof4ik

## Summary of change

- created Profile component which represents a form for displaying and updating profile info and avatar
- added updateProfile actions and reducers

profileq1 jpeg

## CHECK LIST
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR

### CODE REVIEW

File: app/src/containers/Profile/Profile.js [hunk 1]
```diff
@@ -1,15 +1,161 @@
...
+    this.setState(nextProps.profile);
+  }
+
+  handleChange = stateName => e => {
+    this.setState({
+      [stateName]: e.target.value
+    });
+  };
+
+  handleSubmit = (e) => {
```
Comment (yurii-kokhalevych) [inline app/src/containers/Profile/Profile.js]: delete ()

File: app/src/containers/Profile/Profile.js [hunk 2]
```diff
@@ -1,15 +1,161 @@
...
+  Tabs, Tab  } from 'react-bootstrap';
 import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import * as profileActions from '../../redux/profileReducers';
+import DateTimeField from 'react-bootstrap-datetimepicker';
+import moment from 'moment';
+
+import './profile.scss';
+
+let FieldGroup = ({id, label, ...props}) => (
```
Comment (yarvanya) [inline app/src/containers/Profile/Profile.js]: Here use "const".

File: app/src/containers/Profile/Profile.js [hunk 3]
```diff
@@ -1,15 +1,161 @@
 import React from 'react';
+import { Row, Col, Image, FormGroup, ControlLabel, Form, FormControl, Button, ButtonToolbar,
+  Tabs, Tab  } from 'react-bootstrap';
 import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import * as profileActions from '../../redux/profileReducers';
+import DateTimeField from 'react-bootstrap-datetimepicker';
+import moment from 'moment';
+
```
Comment (yarvanya) [inline app/src/containers/Profile/Profile.js]: Delete empty space.

File: app/src/containers/Profile/Profile.js [hunk 4]
```diff
@@ -1,15 +1,161 @@
...
+      last_name: this.props.profile.last_name,
+      birth_date: this.props.profile.birth_date,
+      address: this.props.profile.address,
+      city: this.props.profile.city,
+      country: this.props.profile.country,
+      avatar: this.props.profile.avatar
+    };
+  }
+  
+
```
Comment (yarvanya) [inline app/src/containers/Profile/Profile.js]: Delete one empty space.

File: app/src/containers/Profile/Profile.js [hunk 5]
```diff
@@ -1,15 +1,161 @@
...
+                      this.textFields(param, fieldsName)
+                  )}
+                </Row>
+                <Row> 
+                  <hr/> 
+                  <ButtonToolbar>
+                    <Button type='submit' bsStyle='primary'>
+                      Save Changes
+                    </Button>
+                    <span> </span>
```
Comment (AlbaMontis) [inline app/src/containers/Profile/Profile.js]: Use css for this.

File: app/src/containers/Profile/Profile.js [hunk 6]
```diff
@@ -1,15 +1,161 @@
...
+    </FormGroup>
+  </div>
+);
 
 class Profile extends React.Component {
+  constructor (props) {
+    super(props);
+
+    this.state = {
+      first_name: this.props.profile.first_name,
```
Comment (AlbaMontis) [inline app/src/containers/Profile/Profile.js]: Maybe just `this.state = this.props.profile` ?

File: app/src/redux/profileReducers.js [hunk 7]
```diff
@@ -38,6 +41,37 @@ const initialState = {
   error: null
 };
 
+export const updateProfile = profile => {
+  
+  return dispatch => {
+    dispatch(updateProfilerRequest());
+    const profileId = sessionStorage.getItem('profileId');
+    return request()
+      .put(API.HOST + API.PORT + '/api/profile/' + profileId)
```
Comment (AlbaMontis) [inline app/src/redux/profileReducers.js]: Use API.URL.

**Review by yurii-kokhalevych — DISMISSED** [review]

**Review by yarvanya — DISMISSED** [review]

**Review by avoytovich — DISMISSED** [review]

**Review by yarvanya — DISMISSED** [review]

**Review by yarvanya — DISMISSED** [review]

**Review by yurii-kokhalevych — DISMISSED** [review]

**Review by avoytovich — DISMISSED** [review]

File: app/src/redux/profileReducers.js [hunk 8]
```diff
@@ -4,13 +4,16 @@ import { API } from './../helper/constants';
...
+const UPDATE_PROFILE_FAILURE = 'UPDATE_PROFILE_FAILURE';
+const UPDATE_PROFILE_SUCCESS = 'UPDATE_PROFILE_SUCCESS';
 
 export const retrieveProfile = userId => {
 
   return dispatch => {
     dispatch(retrieveProfilerRequest());
     return request()
-      .get(API.HOST + API.PORT + '/api/user/' + userId)
+      .get(API.URL + '/api/user/' + userId)
```
Comment (4asof4ik) [inline app/src/redux/profileReducers.js]: use `${} ...... ${userId}`

File: app/src/redux/profileReducers.js [hunk 9]
```diff
@@ -38,6 +41,37 @@ const initialState = {
   error: null
 };
 
+export const updateProfile = profile => {
+  
+  return dispatch => {
+    dispatch(updateProfilerRequest());
+    const profileId = sessionStorage.getItem('profileId');
+    return request()
+      .put(API.URL + '/api/profile/' + profileId)
```
Comment (4asof4ik) [inline app/src/redux/profileReducers.js]: and here

File: app/src/containers/Profile/Profile.js [hunk 10]
```diff
@@ -1,15 +1,153 @@
...
+
+    this.state = this.props.profile;
+  }
+  
+  componentWillMount() {
+    const { profile } = this.props;
+    this.setState(profile);
+  }
+
+  componentWillReceiveProps(nextProps){
```
Comment (4asof4ik) [inline app/src/containers/Profile/Profile.js]: space before {

File: app/src/containers/Profile/Profile.js [hunk 11]
```diff
@@ -1,15 +1,153 @@
...
+
+  handleChange = stateName => e => {
+    this.setState({
+      [stateName]: e.target.value
+    });
+  };
+
+  handleSubmit = e => {
+    this.props.actions.updateProfile(this.state);
+  }
```
Comment (4asof4ik) [inline app/src/containers/Profile/Profile.js]: ;

fix linter

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

File: app/src/containers/Profile/Profile.js [hunk 12]
```diff
@@ -1,15 +1,149 @@
...
 
 class Profile extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = this.props.profile;
+  }
+  
+  componentWillMount() {
+    const {profile} = this.props;
```
Comment (4asof4ik) [inline app/src/containers/Profile/Profile.js]: you dont need this const here

File: app/src/redux/profileReducers.js [hunk 13]
```diff
@@ -38,6 +41,37 @@ const initialState = {
...
+  
+  return dispatch => {
+    dispatch(updateProfilerRequest());
+    const profileId = sessionStorage.getItem('profileId');
+    return request()
+      .put(API.URL + `/api/profile/${profileId}`)
+      .send(profile)
+      .type('json')
+      .end((error, res) => {
+        if (error) {
```
Comment (4asof4ik) [inline app/src/redux/profileReducers.js]: improve

File: app/src/redux/profileReducers.js [hunk 14]
```diff
@@ -38,6 +41,37 @@ const initialState = {
   error: null
 };
 
+export const updateProfile = profile => {
+  
+  return dispatch => {
+    dispatch(updateProfilerRequest());
+    const profileId = sessionStorage.getItem('profileId');
+    return request()
+      .put(API.URL + `/api/profile/${profileId}`)
```
Comment (4asof4ik) [inline app/src/redux/profileReducers.js]: use `${}` for all

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

File: app/src/containers/Profile/Profile.js [hunk 15]
```diff
@@ -1,15 +1,148 @@
...
+
+    return(
+      <div className='profile-details'>
+        <Col sm={8} className='container'>
+          <Tabs defaultActiveKey={1} id='uncontrolled-tab'>
+            <Tab eventKey={1} title='Profile Info'>
+              <h1>Edit Profile</h1>
+              <Form horizontal onSubmit={this.handleSubmit}>
+                <Row>
+                  <Col  md={4} className='text-center'>
```
Comment (yarvanya) [inline app/src/containers/Profile/Profile.js]: Delete one space.

File: app/src/containers/Profile/Profile.js [hunk 16]
```diff
@@ -1,15 +1,148 @@
...
+                    <FormGroup key='avatar'>
+                      <Image id='img-circle-avatar' src={profile.avatar} circle/>
```
Comment (yarvanya) [inline app/src/containers/Profile/Profile.js]: Space before  />

File: app/src/containers/Profile/Profile.js [hunk 17]
```diff
@@ -1,15 +1,148 @@
...
+      <ControlLabel>{label}</ControlLabel>
+      <FormControl {...props} />
+    </FormGroup>
+  </div>
+);
 
 class Profile extends React.Component {
+  constructor(props) {
+    super(props);
+
```
Comment (yarvanya) [inline app/src/containers/Profile/Profile.js]: Delete this empty blank.

File: app/src/redux/eventsReducers.js [hunk 18]
```diff
@@ -1,26 +1,22 @@
+import  { API }  from './../helper/constants';
```
Comment (4asof4ik) [inline app/src/redux/eventsReducers.js]: Api request from ./helper

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

File: app/src/redux/eventsReducers.js (hunk 18 above)
Comment (yuliannakorol) [reply app/src/redux/eventsReducers.js]: it won't work because /helper/index.js is not yet merged on dev

**Review by yarvanya — DISMISSED** [review]

**Review by yarvanya — DISMISSED** [review]

**Review by avoytovich — DISMISSED** [review]

**Review by yurii-kokhalevych — DISMISSED** [review]

**Comment by lhalam** [comment]
Conflicting files
app/src/redux/eventsReducers.js
app/src/redux/login.js

**Review by avoytovich — APPROVED** [review]

**Review by yurii-kokhalevych — APPROVED** [review]

**Review by 4asof4ik — APPROVED** [review]

**Pull request merged**